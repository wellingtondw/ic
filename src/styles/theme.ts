export default {
  border: {
    radius: '0.625rem'
  },
  font: {
    main: {
      family: "'Montserrat', sans-serif",
      normal: 400,
      bold: 700
    },
    secondary: {
      family: "'Lato', sans-serif",
      bold: 700
    },
    sizes: {
      xsmall: '0.75rem',
      small: '0.875rem',
      normal: '1rem',
      medium: '1.125rem',
      large: '2.25rem',
      xlarge: '4.5rem'
    }
  },
  colors: {
    primary: '#2C97D1',
    black: '#2A2A2A',
    blackBg: '#2B2B2B',
    white: '#FFFFFF',
    yellow: '#FBFE63'
  },
  spacings: {
    xxxsmall: '0.25rem', // 4px
    xxsmall: '0.5rem', //8px
    xsmall: '1rem', //16px
    small: '1.5rem', //24px
    medium: '2rem', // 32px
    large: '2.5rem', //40px
    xlarge: '3rem', //48px
    xxlarge: '3.5rem' //56px
  },
  transition: {
    default: '0.3s ease-in-out',
    fast: '0.1s ease-in-out'
  }
} as const
