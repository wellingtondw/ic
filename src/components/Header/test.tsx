import { screen, fireEvent } from '@testing-library/react'
import { renderWithTheme } from '../../utils/tests/helpers'

import { Header } from '.'
import theme from '../../styles/theme'

const mockedNavigate = jest.fn()

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockedNavigate
  }
})

describe('<Header />', () => {
  it('should be able to render correctly', () => {
    const { container } = renderWithTheme(<Header />)

    expect(container.firstChild).toHaveStyleRule(
      'padding',
      `0 ${theme.spacings.medium}`,
      {
        media: '(min-width: 480px)'
      }
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be able to back in history', () => {
    renderWithTheme(<Header />)

    const buttonEl = screen.getByRole('button', { name: /back/i })

    fireEvent.click(buttonEl)

    expect(mockedNavigate).toHaveBeenCalledTimes(1)
    expect(mockedNavigate).toHaveBeenCalledWith(-1)
  })
})
