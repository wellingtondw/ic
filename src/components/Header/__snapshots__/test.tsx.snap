// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should be able to render correctly 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c1 {
  height: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  border: 0;
  background: transparent;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  color: #2A2A2A;
}

.c1 > svg {
  width: 32px;
  height: 32px;
  margin-right: 0.25rem;
}

.c1:hover:not(:disabled) {
  color: #222;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 4rem;
  padding: 0 1rem;
}

@media (min-width:480px) {
  .c0 {
    padding: 0 2rem;
  }
}

<header
  class="c0"
>
  <button
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="c2"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 8a.5.5 0 0 1-.5.5H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5a.5.5 0 0 1 .5.5z"
        fill-rule="evenodd"
      />
    </svg>
    back
  </button>
</header>
`;
